models/zuora/zuora_subscription.sql

```diff
@@ -1,12 +1,158 @@
+with sub_mrr as (
 select
-    id                                  as subscr_id,
-    status                              as subscr_status,
-    termtype                            as subscr_term_type,
-    accountid                           as account_id,
-    contracteffectivedate::timestamp    as subscr_start,
-    subscriptionenddate::timestamp      as subscr_end,
-    name                                as subscr_name,
-    "version#392c30e6081c24fb78ddf6d622de4f33"::integer 
-                                        as subscr_version,
-    *
-from zuora.zuora_subscription
+  c.id as sub_id,
+  sum(cast(a.mrr as decimal(18,2))) as mrr,
+  sum(cast(a.tcv as decimal(18,2))) as tcv
+from zuora.zuora_rate_plan_charge a
+join zuora.zuora_rate_plan b
+on a.rateplanid = b.id
+join zuora.zuora_subscription c
+on b.subscriptionid = c.id
+where a.islastsegment = TRUE
+group by 1
+),
+
+sub_billing_periods AS (
+SELECT 
+  listagg(billingperiod,', ') WITHIN GROUP (ORDER BY billingperiod) AS billing_periods,
+  subscriptionid
+FROM (
+SELECT
+  DISTINCT b.subscriptionid,a.billingperiod
+FROM zuora.zuora_rate_plan_charge a
```

Comment:
@dwallacerjm If you need to reference the same base model several times, just create a CTE at the beginning of the query so that you hit the underlying table once. For example,

`with
rate_plan_charge as
(
select *
from {{env.schema}}.zuora_rate_plan_charge
)
`

then you can simply do `select from rate_plan_charge`